src/Rector/Property/ProphecyPHPDocRector.php

```diff
@@ -0,0 +1,184 @@
+
+namespace Rector\PHPUnit\Rector\Property;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\Node\Stmt\Use_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\Rector\Property\ProphecyPHPDocRector\ProphecyPHPDocRectorTest
+ */
+class ProphecyPHPDocRector extends AbstractRector
+{
+    public function __construct(
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Add correct @var to ObjectProphecy instances based on $this->prophesize() call.',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class HelloTest extends TestCase
+{
+    /**
+     * @var SomeClass
+     */
+    private $test;
+
+    public function setUp(): void
+    {
+        $this->test = $this->prophesize(SomeClass::class);
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class HelloTest extends TestCase
+{
+    /**
+     * @var ObjectProphecy<SomeClass>
+     */
+    private $test;
+
+    public function setUp(): void
+    {
+        $this->test = $this->prophesize(SomeClass::class);
+    }
+}
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        // PhpParser\Node\Expr\Assign
+        // PhpParser\Node\Expr\MethodCall
+        return [Assign::class];
+    }
+
+    public function refactor(Node $node)
+    {
+        if (! $this->testsNodeAnalyzer->isInTestClass($node)) {
+            return null;
+        }
+
+        if (! $node instanceof Assign) {
+            return;
+        }
+
+        $expr = $node->expr;
+
+        if (! $expr instanceof MethodCall) {
+            return;
+        }
+
+        $var = $expr->var;
+        if (! $var instanceof Variable) {
+            return;
+        }
+
+        if ($var->name !== 'this') {
+            return;
+        }
+
+        $name = $expr->name;
+        if (! $name instanceof Identifier) {
+            return;
+        }
+
+        if ($name->name !== 'prophesize') {
+            return;
+        }
+
+        $value = $expr->args[0]->value;
+
+        if ($value instanceof String_) {
+            $prophesizeClassParts = \explode('\\', $value->value);
+        } elseif ($value instanceof ClassConstFetch) {
+            $prophesizeClassParts = $value->class->parts;
```

Comment:
While appling the rule on my codebase: This current fails in case of an alias import of the class e.g.:

```php
use Some\A as B;
use Some\Other\A;

$this->test = $this->prophesize(B::class);
```

Ends in:

```
/** @var ObjectProphecy<A> */
private $test;
```

Should be:

```
/** @var ObjectProphecy<B> */
private $test;
```

test required for this case.

Reply:
Same here, always use FQN class name.